presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -578,10 +578,19 @@ private DynamicFiltersResult createDynamicFilters(JoinNode node, List<JoinNode.E
                 // Even if equiJoinClauses.equals(node.getCriteria), current dynamic filters may not match equiJoinClauses
                 ImmutableMap.Builder<String, Symbol> dynamicFiltersBuilder = ImmutableMap.builder();
                 ImmutableList.Builder<Expression> predicatesBuilder = ImmutableList.builder();
+                Map<Symbol, String> buildSymbolToDynamicFilter = node.getDynamicFilters().entrySet().stream()
+                        .collect(toMap(Map.Entry::getValue, Map.Entry::getKey));
                 for (JoinNode.EquiJoinClause clause : equiJoinClauses) {
                     Symbol probeSymbol = clause.getLeft();
                     Symbol buildSymbol = clause.getRight();
-                    String id = "df_" + idAllocator.getNextId().toString();
+                    String id;
+                    if (buildSymbolToDynamicFilter.containsKey(buildSymbol)) {
+                        id = buildSymbolToDynamicFilter.get(buildSymbol);
+                    }
+                    else {
+                        id = "df_" + idAllocator.getNextId().toString();
+                        buildSymbolToDynamicFilter.put(buildSymbol, id);
+                    }
```

Comment:
```suggestion
                    id = buildSymbolToDynamicFilter.computeIfAbsent(buildSymbol, key -> "df_" + idAllocator.getNextId().toString());
```

However, it's not a good idea to modify the map returned by `.collect(toMap(...))`. There's no guarantee that the map is mutable or that it's safe to change. From the javadocs:

> There are no guarantees on the type, mutability, serializability, or thread-safety of the Map returned.